File: backoffice/src/shared/domain/DomainException.py
"""
 *
 * Classes 
 *
"""

class DomainException( Exception ):

    """
     *
     * Parameters 
     *
    """

    __code : int

    """
     *
     * Methods 
     *
    """

    def __init__( self, code : int, message : str ) -> None:
        super().__init__( message )
        self.__code = code

    def code( self ) -> int:
        return self.__code

File: attention/src/shared/domain/exception/DomainException.cs
/* 
 * 
 * Classes 
 *
*/

namespace attention.src.shared.domain {

    public class DomainException : Exception {

        /* 
         * 
         * Constants
         *
        */

        public const int SERVER_INTERNAL_ERROR = 103;

        /* 
         * 
         * Attributes
         *
        */

        private int _code;

        /* 
         * 
         * Methods
         *
        */

        public DomainException( int code, string message ) : base( message ) {
            _code = code;
        }

        public int code() {
            return _code;
        }

    }

}